### PR TITLE
--unanimous tests (contains all commits from, and supersedes, #35)

### PR DESCRIPTION
**Pull request by rwaldron**
- Combines #35 with necessary fixes to:
  -  restore default behavior
  - ensure correct conditions for `--unanimous`
- Adds tests (adds spidermonkey install to travis.yml)

### CODE REVIEW

**Comment by rwaldron** [comment]
I'm still working out kinks in travis, so this is not yet ready for merging

File: lib/Reporter.js [hunk 1]
```diff
@@ -1,16 +1,34 @@
...
+    }
+    return true;
+  }
+
+  static coalesceInto(records, host, result, resultsMap /*{result:[...hosts]}*/, sep = "\n") {
     let found = false;
 
-    for (let row of table) {
-      if (row[1] === result) {
+    for (let record of records) {
```
Comment (rwaldron) [inline lib/Reporter.js]: This might not always be a "table", and the default reporter shouldn't be talking about tables anyway ;)

Comment (bterlson) [inline lib/Reporter.js]: You don't need to justify your style fixes :-P

**Comment by bterlson** [comment]
Travis worked wonderfully, btw: https://travis-ci.org/bterlson/eshost-cli/builds/267687347

**Comment by dilijev** [comment]
@rwaldron Sorry for dragging my feet on the tests. Thanks for the updates and for adding the tests!

**Pull request merged**